internal/hostname/fqdn_nix.go

```diff
@@ -0,0 +1,28 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// This file is exactly pulled from datadog-agent/pkg/util/hostname
+
+//go:build !windows
+// +build !windows
+
+package hostname
+
+import (
+	"context"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+func getSystemFQDN() (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*1)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, "/bin/hostname", "-f")
```

Comment:
`/bin/hostname -f` is not reliable in containerized environments. E.g in k8s it returns the pod name, in docker it returns the container ID.

Reply:
This is "OK" since it's a best-effort attempt to determine the hostname, we will only use this hostname if we can't reach any cloud APIs

Reply:
I understand that we're doing this in best-effort, at the same time we should be careful about the correctness of the values that we're going to collect and send.

I think we should decide what "best-effort" means in this context, is it just about supporting a subset of environments / hostname providers or does it also include the correctness of reported values?
IMHO we should avoid reporting incorrect values, but if this is expected/desired by the product owners we should document it thoroughly in the code and make it super clear for the teams who will process the values.

Reply:
Hmm these are good questions. A follow-up here: Is there a reliable way for us to know we're in a containerized environment (without asking the user to configure something like the agent requires?).
I will also reach out to the PM here to see what is expected here.

Reply:
> Is there a reliable way for us to know we're in a containerized environment (without asking the user to configure something like the agent requires?).

I'm afraid there isn't a reliable way to tell 😐 Perhaps worth asking the containers team?

And thanks for reaching out to the product team!